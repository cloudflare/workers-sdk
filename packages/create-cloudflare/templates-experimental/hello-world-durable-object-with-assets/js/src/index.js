import { DurableObject } from "cloudflare:workers";

/**
 * Welcome to Cloudflare Workers! This is your first Durable Objects application.
 *
 * - Run `npm run dev` in your terminal to start a development server
 * - Open a browser tab at http://localhost:8787/ to see your Durable Object in action
 * - Run `npm run deploy` to publish your application
 *
 * Learn more at https://developers.cloudflare.com/durable-objects
 */

/**
 * Env provides a mechanism to reference bindings declared in wrangler.toml within JavaScript
 *
 * @typedef {Object} Env
 * @property {DurableObjectNamespace} MY_DURABLE_OBJECT - The Durable Object namespace binding
 */

/** A Durable Object's behavior is defined in an exported Javascript class */
export class MyDurableObject extends DurableObject {
	/**
	 * The constructor is invoked once upon creation of the Durable Object, i.e. the first call to
	 * 	`DurableObjectStub::get` for a given identifier (no-op constructors can be omitted)
	 *
	 * @param {DurableObjectState} ctx - The interface for interacting with Durable Object state
	 * @param {Env} env - The interface to reference bindings declared in wrangler.toml
	 */
	constructor(ctx, env) {
		super(ctx, env);
	}

	/**
	 * The Durable Object exposes an RPC method sayHello which will be invoked when when a Durable
	 *  Object instance receives a request from a Worker via the same method invocation on the stub
	 *
	 * @param {string} name - The name provided to a Durable Object instance from a Worker
	 * @returns {Promise<string>} The greeting to be sent back to the Worker
	 */
	async sayHello(name) {
		return `Hello, ${name}!`;
	}
}

export default {
	/**
	 * This is the standard fetch handler for a Cloudflare Worker
	 *
	 * @param {Request} request - The request submitted to the Worker from the client
	 * @param {Env} env - The interface to reference bindings declared in wrangler.toml
	 * @param {ExecutionContext} ctx - The execution context of the Worker
	 * @returns {Promise<Response>} The response to be sent back to the client
	 */
	async fetch(request, env, ctx) {
		// We will create a `DurableObjectId` using the pathname from the Worker request
		// This id refers to a unique instance of our 'MyDurableObject' class above
		let id = env.MY_DURABLE_OBJECT.idFromName(new URL(request.url).pathname);

		// This stub creates a communication channel with the Durable Object instance
		// The Durable Object constructor will be invoked upon the first call for a given id
		let stub = env.MY_DURABLE_OBJECT.get(id);

		// We call the `sayHello()` RPC method on the stub to invoke the method on the remote
		// Durable Object instance
		let greeting = await stub.sayHello("world");

		return new Response(greeting);
	},
};
