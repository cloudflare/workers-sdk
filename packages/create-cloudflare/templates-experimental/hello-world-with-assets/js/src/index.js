/**
 * Welcome to Cloudflare Workers! This is your first worker.
 *
 * - Run `npm run dev` in your terminal to start a development server
 * - Open a browser tab at http://localhost:8787/ to see your worker in action
 * - Run `npm run deploy` to publish your worker
 *
 * Learn more at https://developers.cloudflare.com/workers/
 */

export default {
	async fetch(request, env, ctx) {
		return new Response('Hello World!');
	},
};
