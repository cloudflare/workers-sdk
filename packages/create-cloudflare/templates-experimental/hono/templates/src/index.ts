import { Hono } from "hono";

const app = new Hono<{ Bindings: CloudflareBindings }>();

app.get("/message", (c) => {
  return c.text("Hello Hono!");
});

export default app;
