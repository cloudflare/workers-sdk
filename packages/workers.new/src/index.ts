// Redirect https://workers.new/<known> requests to IDE.
// Redirect https://workers.new/*? requests to the Workers Playground.
// Similar to the concept of https://docs.new.
type Redirects = Record<
	string,
	[
		/* subdirectory */ string,
		/* file */ string,
		/* title */ string,
		/* terminal */ string?,
	]
>;

// stackblitz repository source
const source = "github/cloudflare/workers-sdk/tree/main/templates";

// deploy with cloudflare source
const src = "https://github.com/cloudflare/workers-sdk/tree/main/templates";

export const redirects: Redirects = {
	"/pages-image-sharing": [
		"pages-image-sharing",
		"src/index.tsx",
		"Image Sharing Website with Pages Functions",
	],
	"/stream/auth/stripe": [
		"stream/auth/stripe",
		"src/index.html",
		"Stream + Stripe Checkout",
	],
	"/worker-durable-objects": [
		"worker-durable-objects",
		"index.js",
		"Workers Durable Objects counter",
	],
	"/d1": ["worker-d1", "src/index.ts", "Workers D1"],
	"/router": ["worker-router", "index.js", "Workers Router"],
	"/typescript": ["worker-typescript", "src/index.ts", "Workers TypeScript"],
	"/websocket": ["worker-websocket", "index.js", "Workers WebSocket"],
	"/example-wordle": [
		"worker-example-wordle",
		"src/index.ts",
		"Workers Wordle example",
	],
	"/example-request-scheduler": [
		"worker-example-request-scheduler",
		"src/index.ts",
		"Workers Request Scheduler",
	],
	"/websocket-durable-objects": [
		"worker-websocket-durable-objects",
		"src/index.ts",
		"Workers WebSocket Durable Objects",
	],
	"/worktop": ["worker-worktop", "src/index.ts", "Workers Worktop"],
	"/pages-functions-cors": [
		"pages-functions-cors",
		"functions/api/_middleware.ts",
		"Pages Functions CORS",
		"dev",
	],
	"/pages-plugin-static-forms": [
		"pages-plugin-static-forms",
		"functions/_middleware.ts",
		"Pages Plugin static forms",
		"dev",
	],
	"/pages-example-forum-app": [
		"pages-example-forum-app",
		"functions/api/code.ts",
		"Pages Example Forum app",
		"dev",
	],
	"/stream/stream-player": [
		"stream/playback/stream-player",
		"src/index.html",
		"Cloudflare Stream Player",
	],
	"/stream/video-js": [
		"stream/playback/video-js",
		"src/index.html",
		"Cloudflare Stream + Video.js",
	],
	"/stream/vidstack": [
		"stream/playback/vidstack",
		"src/index.html",
		"Cloudflare Stream + Vidstack",
	],
	"/stream/hls-js": [
		"stream/playback/hls-js",
		"src/index.html",
		"Cloudflare Stream + hls.js",
	],
	"/stream/dash-js": [
		"stream/playback/dash-js",
		"src/index.html",
		"Cloudflare Stream + dash.js",
	],
	"/stream/shaka-player": [
		"stream/playback/shaka-player",
		"src/index.js",
		"Cloudflare Stream + Shaka Player",
	],
	"/stream/direct-creator-uploads": [
		"stream/upload/direct-creator-uploads",
		"src/index.html",
		"Direct Creator Uploads to Cloudflare Stream",
	],
	"/stream/direct-creator-uploads-tus": [
		"stream/upload/direct-creator-uploads-tus",
		"src/index.html",
		"Direct Creator Uploads to Cloudflare Stream, using TUS",
	],
	"/stream/webrtc": [
		"stream/webrtc",
		"src/index.html",
		"Stream live video (using WHIP) and playback (using WHEP) over WebRTC with Cloudflare Stream",
	],
	"/stream/webrtc-whip": [
		"stream/webrtc",
		"src/index.html",
		"Stream live video (using WHIP) over WebRTC with Cloudflare Stream",
	],
	"/stream/webrtc-whep": [
		"stream/webrtc",
		"src/index.html",
		"Play live video (using WHEP) over WebRTC with Cloudflare Stream",
	],
	"/stream/signed-urls-public-content": [
		"stream/auth/signed-urls-public-content",
		"src/index.html",
		"Example of using Cloudflare Stream Signed URLs with public content",
	],
};

const worker: ExportedHandler = {
	fetch(request) {
		const { pathname } = new URL(request.url);

		if (pathname === "/templates") {
			return new Response(getListHTML(redirects), {
				headers: { "content-type": "text/html" },
			});
		}

		const redirectUrl = getRedirectUrlForPathname(pathname);
		if (redirectUrl) {
			return Response.redirect(redirectUrl, 302);
		}

		if (pathname === "/python") {
			return Response.redirect(
				"https://workers.cloudflare.com/playground/python",
				302
			);
		} else {
			return Response.redirect(
				"https://workers.cloudflare.com/playground",
				302
			);
		}
	},
};

function getRedirectUrlForPathname(pathname: string): string | undefined {
	const [subdir, file, title, terminal] = redirects[pathname] || [];
	if (subdir) {
		const focus = encodeURIComponent(file);
		const target = `https://stackblitz.com/fork/${source}/${subdir}?file=${focus}&title=${title}&terminal=${
			terminal || "start-stackblitz"
		}`;
		return target;
	}
}

function getListHTML(redirectsParam: Redirects) {
	return /*html*/ `
<html>
<head>
	<link href='https://fonts.googleapis.com/css2?family=Inter:wght@200;300;500;700&display=swap' rel='stylesheet'>
</head>
<style>
	body {
		margin: 2rem 10rem;
		font-family: "Inter", sans-serif;
	}
	h1{
		text-align: center;
		margin: 20px 0;
		font-size: 5rem;
	}

	.nav {
		display: flex;
		justify-content: space-between;
		font-size: 18px;
	}

	a {
		text-decoration: none;
	}

	a {
		color: inherit;
	}

	.heading {
		font-size: 25px;
		text-align: center;
		margin: 0;
		font-weight: 700;
	}

	.subheading {
		text-align: center;
		font-size: 18px;
		font-weight: 300;
	}

	.title {
		font-weight: 400;
		margin-bottom: 50px;
	}

	ul {
		list-style: none;
		display: flex;
    flex-wrap: wrap;
    gap: 20px;
    justify-content: center;
		text-align: center;
		padding-top: 30px;
	}

	li {
		padding: 20px 20px;
		border: 1px solid #d5d7d8;
		border-radius: 10px;
	}

	li:nth-child(-n+3) {
		padding: 20px 20px;
		border: 4px solid #f1740a;
		border-radius: 10px;
		background: #fef1e6;
	}

	.btn {
		margin: 0 10px;
		text-decoration: underline;
	}

	.link {
		padding: 10px;
		border: 1px dotted #f1740a;
		border-radius: 5px;
	}

	.link:hover {
		background: #f1740a;
		color: #fff;
	}

	.card-title {
		font-size: 18px;
		font-weight: 600;
	}

	.url {
		color: #f1740a;
		font-weight: 700;
	}

</style>
<body>
	<nav class="nav">
		<a href="https://workers.cloudflare.com">
			<img src="https://imagedelivery.net/T24Zz2DP7HaLOEAdMToO-g/70363224-4bee-4f48-a775-06ffdfbc6d00/public" height="50" alt="Cloudflare Workers" />
		</a>
		<a href="https://developers.cloudflare.com/workers">Documentation</a>
	</nav>
	<p class="heading">Cloudflare Workers Templates</p>
	<p class="subheading">Ready to use templates to start building applications on Cloudflare Workers.</p>
	<ul>
		${Object.keys(redirectsParam)
			.map((pathname) => {
				const [subdir, _file, title, _terminal] =
					redirectsParam[pathname] || [];
				return `<li>
				<p class="card-title"> ${title} </p>
				<p class="title">${subdir}</p>
				<span class="link"><a target="_blank" href="https://workers.new${pathname}">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=${src}/${subdir}">Deploy with Workers</a></span>
				</li>`;
			})
			.join("\n")}
	</ul>
	<p class="subheading">Want to contribute a template? <a class="url" href="https://github.com/cloudflare/workers-sdk/tree/main/templates"> Send a PR to the Wrangler repository.</a></p>
</body>
</html>
`;
}

export default worker;
