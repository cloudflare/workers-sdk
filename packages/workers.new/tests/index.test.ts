import { SELF } from "cloudflare:test";
import { describe, expect, it } from "vitest";
import { redirects } from "../src/index";

describe("worker", () => {
	it("python workers", async () => {
		const resp = await SELF.fetch(`https://workers.new/python`, {
			redirect: "manual",
		});

		const location = resp.headers.get("Location");

		expect(location).toMatchInlineSnapshot(
			'"https://workers.cloudflare.com/playground/python"'
		);
	});

	it("templates list", async () => {
		const resp = await SELF.fetch(`https://workers.new/templates`, {
			redirect: "manual",
		});

		expect(await resp.text()).toMatchSnapshot();
	});

	it("static definitions", async () => {
		const redirectMapping = new Map();
		for (const [path] of Object.entries(redirects)) {
			const resp = await SELF.fetch(`https://workers.new${path}`, {
				redirect: "manual",
			});

			const location = resp.headers.get("Location");

			redirectMapping.set(path, location);
		}
		expect(redirectMapping).toMatchInlineSnapshot(`
			Map {
			  "/pages-image-sharing" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/pages-image-sharing?file=src%2Findex.tsx&title=Image%20Sharing%20Website%20with%20Pages%20Functions&terminal=start-stackblitz",
			  "/stream/auth/stripe" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/auth/stripe?file=src%2Findex.html&title=Stream%20+%20Stripe%20Checkout&terminal=start-stackblitz",
			  "/worker-durable-objects" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-durable-objects?file=index.js&title=Workers%20Durable%20Objects%20counter&terminal=start-stackblitz",
			  "/d1" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-d1?file=src%2Findex.ts&title=Workers%20D1&terminal=start-stackblitz",
			  "/router" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-router?file=index.js&title=Workers%20Router&terminal=start-stackblitz",
			  "/typescript" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-typescript?file=src%2Findex.ts&title=Workers%20TypeScript&terminal=start-stackblitz",
			  "/websocket" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-websocket?file=index.js&title=Workers%20WebSocket&terminal=start-stackblitz",
			  "/example-wordle" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-example-wordle?file=src%2Findex.ts&title=Workers%20Wordle%20example&terminal=start-stackblitz",
			  "/example-request-scheduler" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-example-request-scheduler?file=src%2Findex.ts&title=Workers%20Request%20Scheduler&terminal=start-stackblitz",
			  "/websocket-durable-objects" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-websocket-durable-objects?file=src%2Findex.ts&title=Workers%20WebSocket%20Durable%20Objects&terminal=start-stackblitz",
			  "/worktop" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/worker-worktop?file=src%2Findex.ts&title=Workers%20Worktop&terminal=start-stackblitz",
			  "/pages-functions-cors" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/pages-functions-cors?file=functions%2Fapi%2F_middleware.ts&title=Pages%20Functions%20CORS&terminal=dev",
			  "/pages-plugin-static-forms" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/pages-plugin-static-forms?file=functions%2F_middleware.ts&title=Pages%20Plugin%20static%20forms&terminal=dev",
			  "/pages-example-forum-app" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/pages-example-forum-app?file=functions%2Fapi%2Fcode.ts&title=Pages%20Example%20Forum%20app&terminal=dev",
			  "/stream/stream-player" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/stream-player?file=src%2Findex.html&title=Cloudflare%20Stream%20Player&terminal=start-stackblitz",
			  "/stream/video-js" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/video-js?file=src%2Findex.html&title=Cloudflare%20Stream%20+%20Video.js&terminal=start-stackblitz",
			  "/stream/vidstack" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/vidstack?file=src%2Findex.html&title=Cloudflare%20Stream%20+%20Vidstack&terminal=start-stackblitz",
			  "/stream/hls-js" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/hls-js?file=src%2Findex.html&title=Cloudflare%20Stream%20+%20hls.js&terminal=start-stackblitz",
			  "/stream/dash-js" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/dash-js?file=src%2Findex.html&title=Cloudflare%20Stream%20+%20dash.js&terminal=start-stackblitz",
			  "/stream/shaka-player" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/playback/shaka-player?file=src%2Findex.js&title=Cloudflare%20Stream%20+%20Shaka%20Player&terminal=start-stackblitz",
			  "/stream/direct-creator-uploads" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/upload/direct-creator-uploads?file=src%2Findex.html&title=Direct%20Creator%20Uploads%20to%20Cloudflare%20Stream&terminal=start-stackblitz",
			  "/stream/direct-creator-uploads-tus" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/upload/direct-creator-uploads-tus?file=src%2Findex.html&title=Direct%20Creator%20Uploads%20to%20Cloudflare%20Stream,%20using%20TUS&terminal=start-stackblitz",
			  "/stream/webrtc" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/webrtc?file=src%2Findex.html&title=Stream%20live%20video%20(using%20WHIP)%20and%20playback%20(using%20WHEP)%20over%20WebRTC%20with%20Cloudflare%20Stream&terminal=start-stackblitz",
			  "/stream/webrtc-whip" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/webrtc?file=src%2Findex.html&title=Stream%20live%20video%20(using%20WHIP)%20over%20WebRTC%20with%20Cloudflare%20Stream&terminal=start-stackblitz",
			  "/stream/webrtc-whep" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/webrtc?file=src%2Findex.html&title=Play%20live%20video%20(using%20WHEP)%20over%20WebRTC%20with%20Cloudflare%20Stream&terminal=start-stackblitz",
			  "/stream/signed-urls-public-content" => "https://stackblitz.com/fork/github/cloudflare/workers-sdk/tree/main/templates/stream/auth/signed-urls-public-content?file=src%2Findex.html&title=Example%20of%20using%20Cloudflare%20Stream%20Signed%20URLs%20with%20public%20content&terminal=start-stackblitz",
			}
		`);
	});
});
