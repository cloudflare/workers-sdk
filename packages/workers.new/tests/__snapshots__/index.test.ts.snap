// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`worker > templates list 1`] = `
"
<html>
<head>
	<link href='https://fonts.googleapis.com/css2?family=Inter:wght@200;300;500;700&display=swap' rel='stylesheet'>
</head>
<style>
	body {
		margin: 2rem 10rem;
		font-family: "Inter", sans-serif;
	}
	h1{
		text-align: center;
		margin: 20px 0;
		font-size: 5rem;
	}

	.nav {
		display: flex;
		justify-content: space-between;
		font-size: 18px;
	}

	a {
		text-decoration: none;
	}

	a {
		color: inherit;
	}

	.heading {
		font-size: 25px;
		text-align: center;
		margin: 0;
		font-weight: 700;
	}

	.subheading {
		text-align: center;
		font-size: 18px;
		font-weight: 300;
	}

	.title {
		font-weight: 400;
		margin-bottom: 50px;
	}

	ul {
		list-style: none;
		display: flex;
    flex-wrap: wrap;
    gap: 20px;
    justify-content: center;
		text-align: center;
		padding-top: 30px;
	}

	li {
		padding: 20px 20px;
		border: 1px solid #d5d7d8;
		border-radius: 10px;
	}

	li:nth-child(-n+3) {
		padding: 20px 20px;
		border: 4px solid #f1740a;
		border-radius: 10px;
		background: #fef1e6;
	}

	.btn {
		margin: 0 10px;
		text-decoration: underline;
	}

	.link {
		padding: 10px;
		border: 1px dotted #f1740a;
		border-radius: 5px;
	}

	.link:hover {
		background: #f1740a;
		color: #fff;
	}

	.card-title {
		font-size: 18px;
		font-weight: 600;
	}

	.url {
		color: #f1740a;
		font-weight: 700;
	}

</style>
<body>
	<nav class="nav">
		<a href="https://workers.cloudflare.com">
			<img src="https://imagedelivery.net/T24Zz2DP7HaLOEAdMToO-g/70363224-4bee-4f48-a775-06ffdfbc6d00/public" height="50" alt="Cloudflare Workers" />
		</a>
		<a href="https://developers.cloudflare.com/workers">Documentation</a>
	</nav>
	<p class="heading">Cloudflare Workers Templates</p>
	<p class="subheading">Ready to use templates to start building applications on Cloudflare Workers.</p>
	<ul>
		<li>
				<p class="card-title"> Image Sharing Website with Pages Functions </p>
				<p class="title">pages-image-sharing</p>
				<span class="link"><a target="_blank" href="https://workers.new/pages-image-sharing">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/pages-image-sharing">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Stream + Stripe Checkout </p>
				<p class="title">stream/auth/stripe</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/auth/stripe">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/auth/stripe">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers Durable Objects counter </p>
				<p class="title">worker-durable-objects</p>
				<span class="link"><a target="_blank" href="https://workers.new/worker-durable-objects">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-durable-objects">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers D1 </p>
				<p class="title">worker-d1</p>
				<span class="link"><a target="_blank" href="https://workers.new/d1">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-d1">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers Router </p>
				<p class="title">worker-router</p>
				<span class="link"><a target="_blank" href="https://workers.new/router">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-router">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers TypeScript </p>
				<p class="title">worker-typescript</p>
				<span class="link"><a target="_blank" href="https://workers.new/typescript">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-typescript">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers WebSocket </p>
				<p class="title">worker-websocket</p>
				<span class="link"><a target="_blank" href="https://workers.new/websocket">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-websocket">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers Wordle example </p>
				<p class="title">worker-example-wordle</p>
				<span class="link"><a target="_blank" href="https://workers.new/example-wordle">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-example-wordle">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers Request Scheduler </p>
				<p class="title">worker-example-request-scheduler</p>
				<span class="link"><a target="_blank" href="https://workers.new/example-request-scheduler">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-example-request-scheduler">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers WebSocket Durable Objects </p>
				<p class="title">worker-websocket-durable-objects</p>
				<span class="link"><a target="_blank" href="https://workers.new/websocket-durable-objects">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-websocket-durable-objects">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Workers Worktop </p>
				<p class="title">worker-worktop</p>
				<span class="link"><a target="_blank" href="https://workers.new/worktop">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/worker-worktop">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Pages Functions CORS </p>
				<p class="title">pages-functions-cors</p>
				<span class="link"><a target="_blank" href="https://workers.new/pages-functions-cors">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/pages-functions-cors">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Pages Plugin static forms </p>
				<p class="title">pages-plugin-static-forms</p>
				<span class="link"><a target="_blank" href="https://workers.new/pages-plugin-static-forms">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/pages-plugin-static-forms">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Pages Example Forum app </p>
				<p class="title">pages-example-forum-app</p>
				<span class="link"><a target="_blank" href="https://workers.new/pages-example-forum-app">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/pages-example-forum-app">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream Player </p>
				<p class="title">stream/playback/stream-player</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/stream-player">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/stream-player">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream + Video.js </p>
				<p class="title">stream/playback/video-js</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/video-js">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/video-js">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream + Vidstack </p>
				<p class="title">stream/playback/vidstack</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/vidstack">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/vidstack">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream + hls.js </p>
				<p class="title">stream/playback/hls-js</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/hls-js">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/hls-js">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream + dash.js </p>
				<p class="title">stream/playback/dash-js</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/dash-js">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/dash-js">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Cloudflare Stream + Shaka Player </p>
				<p class="title">stream/playback/shaka-player</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/shaka-player">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/playback/shaka-player">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Direct Creator Uploads to Cloudflare Stream </p>
				<p class="title">stream/upload/direct-creator-uploads</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/direct-creator-uploads">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/upload/direct-creator-uploads">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Direct Creator Uploads to Cloudflare Stream, using TUS </p>
				<p class="title">stream/upload/direct-creator-uploads-tus</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/direct-creator-uploads-tus">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/upload/direct-creator-uploads-tus">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Stream live video (using WHIP) and playback (using WHEP) over WebRTC with Cloudflare Stream </p>
				<p class="title">stream/webrtc</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/webrtc">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/webrtc">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Stream live video (using WHIP) over WebRTC with Cloudflare Stream </p>
				<p class="title">stream/webrtc</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/webrtc-whip">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/webrtc">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Play live video (using WHEP) over WebRTC with Cloudflare Stream </p>
				<p class="title">stream/webrtc</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/webrtc-whep">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/webrtc">Deploy with Workers</a></span>
				</li>
<li>
				<p class="card-title"> Example of using Cloudflare Stream Signed URLs with public content </p>
				<p class="title">stream/auth/signed-urls-public-content</p>
				<span class="link"><a target="_blank" href="https://workers.new/stream/signed-urls-public-content">Open with StackBlitz</a></span>
				<span class="link"><a target="_blank" href="https://deploy.workers.cloudflare.com/?url=https://github.com/cloudflare/workers-sdk/tree/main/templates/stream/auth/signed-urls-public-content">Deploy with Workers</a></span>
				</li>
	</ul>
	<p class="subheading">Want to contribute a template? <a class="url" href="https://github.com/cloudflare/workers-sdk/tree/main/templates"> Send a PR to the Wrangler repository.</a></p>
</body>
</html>
"
`;
